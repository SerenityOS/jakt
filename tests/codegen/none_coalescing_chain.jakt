/// Expect:
/// - output: "1\n7\n3\n6\n"

fn main() {
    let my_first_none: i64? = None
    let my_second_none: i64? = None

    let my_first_some: i64? = 6
    let my_second_some: i64? = 7

    let a = my_first_none ?? my_second_none ?? 1
    let b = my_first_none ?? my_second_some ?? 2
    let c = my_first_none ?? my_second_none ?? my_second_none ?? my_second_none ?? 3
    let d = my_first_some ?? my_second_some ?? 4

    println("{}", a)
    println("{}", b)
    println("{}", c)
    println("{}", d)
}
