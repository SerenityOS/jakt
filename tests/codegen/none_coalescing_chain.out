1
7
3
6
